crates/bevy_window/src/windows.rs

```diff
@@ -62,16 +65,36 @@ impl Windows {
     }
 
     /// An iterator over all registered [`Window`]s.
-    pub fn iter(&self) -> impl Iterator<Item = &Window> {
+    pub fn iter(&self) -> Values<'_, WindowId, Window> {
```

Comment:
This change is unfortunate. 

I don't think we should be exposing the hashbrown iterators here as public api. OTOH I don't have a better solution

Reply:
Can't we still use a trait impl here? 
```suggestion
    pub fn iter(&self) -> impl Iterator<Item=&'_ Window> + '_ {
```

Reply:
The reason we don't is because this return type is used for the `IntoIterator` implementation below.

Perhaps the `IntoIterator` implementation could be the source of truth, however? Then this could at least maintain the `impl Trait`